src/lib/profiles/data-management/Current/LoggingManagement.cpp

```diff
@@ -475,59 +447,57 @@ WEAVE_ERROR LoggingManagement::SetExchangeManager(nl::Weave::WeaveExchangeManage
  * @brief
  *   LoggingManagement constructor
  *
- * For prioritization to work correctly, inBuffers must be incrementally
- * increasing in priority.
+ * Initialize the LoggingManagement with an array of LogStorageResources.  The
+ * array must provide a resource for each valid importance level; the first
+ * element in the array corresponds to ImportanceType::ProductionCritical and
+ * the last element of the array corresponds to ImportanceType::kImportanceType_Last.
  *
  * @param[in] inMgr         WeaveExchangeManager to be used with this logging subsystem
  *
- * @param[in] inNumBuffers  Number of buffers to use for event storage
- *
- * @param[in] inBufferLengths Description of inBufferLengths
+ * @param[in] inLogStorageResources  An array of LogStorageResources for each importance level.
  *
- * @param[in] inBuffers     The buffers to use for actual event logging.
- *
- * @param[in] inCounterKeys Keys naming persisted counters
- *
- * @param[in] inCounterEpochs An array of epochs for each of the persisted counters
- *
- * @param[in] inCounterStorage Application-provided storage for the persistent counters
- *
- * @return LoggingManagement
  */
-LoggingManagement::LoggingManagement(WeaveExchangeManager * inMgr, size_t inNumBuffers, size_t * inBufferLengths, void ** inBuffers,
-                                     nl::Weave::Platform::PersistedStorage::Key * inCounterKeys, const uint32_t * inCounterEpochs,
-                                     nl::Weave::PersistedCounter ** inCounterStorage)
+
+LoggingManagement::LoggingManagement(nl::Weave::WeaveExchangeManager * inMgr,
```

Comment:
This implementation assumes that number of input buffers is always kImportanceType_Last, which is not always correct. Some platform configurations can disable Info and Debug buffers.
size_t inNumBuffers   should be added as an input to this function.

Reply:
Agreed, I'm adding that to the PR.

Reply:
done